gst/nnstreamer/elements/gsttensor_trainer.c

```diff
@@ -240,6 +243,11 @@ gst_tensor_trainer_class_init (GstTensorTrainerClass * klass)
           G_PARAM_READWRITE | GST_PARAM_MUTABLE_READY |
           G_PARAM_STATIC_STRINGS));
 
+  g_object_class_install_property (gobject_class, PROP_STOP_MODEL_TRAINING,
+      g_param_spec_boolean ("stop-model-training", "Stop model training",
+          "Model training in progress is terminated.", FALSE,
+          G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS));
```

Comment:
1. The description and its behavior doesn't match.
2. What are the possible states? Is it really ok to have "not terminated" and "terminated"? Wouldn't it be "ready", "running", "paused", "terminated", ... ?
3. You are calling gst_tensor_trainer_stop_model_training() unconditionally. And, it is not consistent with the description.

Reply:
Thank you, I will update.